qiskit/algorithms/optimizers/cmaes.py

```diff
@@ -0,0 +1,400 @@
+    sigma: float
+
+    # Parameters for the stoping criteria
+    _best_func_hist: List[float]
+    _median_func_hist: List[float]
+    _worst_func_eval: Optional[float]
+
+
+class CMAES(SteppableOptimizer):
+    """
+    Covariance Matrix Adaptation Evolution Strategy minimization routine.
+    This Optimizer is based on the paper: `arXiv:1604.00772 <https://arxiv.org/abs/1604.00772>`_.
+    CMA-ES is an evolution strategy based on sampling points from a multivariate normal distribution
+    with a given centroid and covariance matrix.
+    From the :math:`\lambda\ points sampled in generation ``g``, a new centroid is calculated as:
+    .. math::
+        x^{g+1} = \sum_{i=1}^{\mu} w_i x_{i:\lambda}^{g}
+        where :math:`\sum_{i=1}^{\mu} w_i = 1` and :math:`x_{i:\lambda}^{g}` are the best :math:`\mu`
+        points from generation ``g`` ordered in function of their objective function image.
+
+    A new covariance matrix is also computed taking into account not only the current population,
+    but also the previous covariance matrix.
+
+    The optimization is stopped either at the begining of the optimization process if the initial
+    parameters were not well chosen, or once the objective function stops decreasing for a given
+    number of generations. There are several criteria that can be manually set by the user, and
+    are well described in the paper.
+
+    One evaluation with a funciton evaluation outlier with a significantly low value is not
+    enough to stop the optimization process or significantly disturb the position of the centroid,
+    and an outlier with a significantly high value would simply be ignored by the algorithm.
+    Therefore, this optimizer is resistant to noise in the objective function and can be used with
+    non-smooth functions.
+    """
+
+    def __init__(
+        self,
+        maxiter: int = 100,
+        callback: Optional[CALLBACK] = None,
+        sigma_0: float = 0.5,
+        weights: Optional[np.ndarray] = None,
+        termination_criteria: Optional[Dict[str, Optional[float]]] = None,
+    ) -> None:
+        """
+        Args:
+            maxiter: Maximum number of iterations.
+            callback: A callback function to be called after each iteration when using :meth:`~.minimize`.
+            sigma_0: Standard deviation for the sampling of the first generation.
+            weights: Weights used to compute the next centroid and covariance matrix. If ``None``, they
+            will take a default value. All components must be positive, arranged in decreasing order,
+            and add up to one.
+            termination_criteria: Dictionary with the terimnation criteria to be used and the tolerances
+            for each of them. The different options and their default values are:
+
+            * NoEffectAxis: 0
+            * NoeffectCoord: 1e4
+            * TolFun: 1e-12
+            * TolX: :math:`\sigma`
```

Comment:
What value is this? I see a parameter called `sigma_0` is it related to that? Update I see now below in the code it is